src/RandomSetGenerator.java

```diff
@@ -0,0 +1,106 @@
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+
+public class RandomSetGenerator {
+
+	static private Random randomNumberGenerator;
+	
+	public RandomSetGenerator() {
+		this(new SecureRandom()); 
+	}
+
+	public RandomSetGenerator(Random generator) {
+		randomNumberGenerator = generator;
+	}
+
+	public List<Integer> generate(int quantity, int maxValue) throws IllegalArgumentException {
+		Set<Integer> randomSet;
+		if(quantity <= 0 || maxValue <= 0 || quantity > maxValue) {
+			throw new IllegalArgumentException("invalid inputs to generate(). Inputs: quantity: " + quantity + ", maxValue: " + maxValue);
```

Comment:
break these out as separate conditionals and make the message descriptive for each failure. you want someone calling your method to know why you threw an exception. also, the quantity can be zero => just return an empty set.